third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -250,20 +253,35 @@ Value convertLayout(int opIdx, ConversionPatternRewriter &rewriter,
          (mDim == 64 && nDim == 4) || (mDim == 4 && nDim == 64));
   auto warpsPerCTA = mfmaLayout.getWarpsPerCTA();
 
-  auto aTensorTy = tensor.getType().cast<MemDescType>();
-  ArrayRef<int64_t> shape = aTensorTy.getShape();
   auto sharedLayout = aTensorTy.getEncoding().cast<SharedEncodingAttr>();
   auto order = sharedLayout.getOrder();
 
   auto elemTy = aTensorTy.getElementType();
   auto kWidth = encoding.getKWidth();
   auto elemsPerInstr = mfmaLayout.getMFMAInstrShapeForOperands(kWidth, opIdx);
-  auto mfmaInstrNonK = elemsPerInstr[nonKDimIdx];
-  auto mfmaInstrK = elemsPerInstr[kDimIdx];
+
+  int64_t mfmaInstrNonK;
+  int64_t mfmaInstrK;
+  // TODO(Lixun): make it simpler
+  // getMFMAInstrShapeForOperands always returns a 2D vector
+  if (rank == 3) {
```

Comment:
I thought we were going to do this: https://github.com/openai/triton/pull/3298#discussion_r1518302848

Reply:
Let's leave this for later,
I am going to enable dot3d for WMMA layout as well, after this I'll try to refactor them uniformly.

Reply:
Are you enabling dot3d for wmma layout in this PR ?

Reply:
No, this will be separate PR.

I my opinion it is easier to review and fix stuff step by step.